base/compiler/abstractinterpretation.jl

```diff
@@ -45,7 +45,11 @@ end
 
 function should_infer_for_effects(sv::InferenceState)
     effects = sv.ipo_effects
-    return is_terminates(effects) && is_effect_free(effects)
+    effects.consistent === ALWAYS_FALSE && return false
+    effects.effect_free === ALWAYS_FALSE && return false
```

Comment:
Should we revisit this condition given semi-concrete?

Reply:
Isn't the effects condition for semi-concrete eval is same as that for concrete eval?